src/TsStateMachinePrinter.ts

```diff
@@ -16,18 +16,32 @@ export default class TsStateMachinePrinter {
   private readonly components: Components;
   private readonly domType: IDomType;
   private readonly buildType: IBuildType;
+  private readonly iterable?: {
+    hasIterable: () => boolean;
+    hasMaplikeSequence: boolean;
+  };
 
-  constructor(i: Types.Interface, printer: Printer, components: Components, options: IOptions) {
+  constructor(
+    i: Types.Interface,
+    printer: Printer,
+    components: Components,
+    options: IOptions,
+    iterable?: {
+      hasIterable: () => boolean;
+      hasMaplikeSequence: boolean;
+    },
```

Comment:
I find this really confusing, specifically how the other classes are passing "iterable" in (this.bodyPrinter.iteratorExtractor). Why not pass these in as options?

Reply:
It felt wrong to pass the whole IteratorExtractor into the state machine printer, so I just defined the properties I wanted available to this class. I could just pass the parameters individually too (I had that initially), but it seemed more bug prone than to leave them on the source object so they all change in unison if some other extractor messed with the underlying properties

Reply:
Hrm... k. I find it much easier to follow the code when it's clear where these iterable methods are coming from... especially since all the this.bodyPrinter.iteratorExtractor is what's being passed in from all the class initializers.

Reply:
I'm fine with changing it. Are you saying we should pass in individual variables though, or the IteratorExtractor?

Reply:
You make a good point about passing them in individually seems more error prone... I would recommend converting `iterable` to `iterableExtractor: IteratorExtractor`.

Reply:
That makes it very clear within TsStateMachinePrinter.ts what I'm looking at, and it minimizes any `this` issues from/if destructuring.

Reply:
Ok, I changed it